### PR TITLE
wine-staging 8.17

### PR DESCRIPTION
**Pull request by hannesbraun**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by hannesbraun** [comment]
I'm not 100% sure about the version. The file contains version 8.17.1. But in the tag and everywhere else, the version is 8.17.

File: Casks/wine-staging.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 cask "wine-staging" do
-  version "8.10"
-  sha256 "4a29553703133095b1ef84cbf840bb227a7b5c0e9c747ee6ac7721599176e33e"
+  version "8.17"
```
Comment (samford) [inline Casks/wine-staging.rb:2]:
```suggestion
  version "8.17.1"
```

File: Casks/wine-staging.rb [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+  sha256 "aa971d9e80d18c65b70d0e863880069a37a1ebd1f15d1e7236aaee830fa88dcb"
 
   # Current winehq packages are deprecated and these are packages from
   # the new maintainers that will eventually be pushed to Winehq.
   # See https://www.winehq.org/pipermail/wine-devel/2021-July/191504.html
-  url "https://github.com/Gcenx/macOS_Wine_builds/releases/download/#{version}/wine-staging-#{version}-osx86.tar.xz",
+  url "https://github.com/Gcenx/macOS_Wine_builds/releases/download/#{version}/wine-staging-#{version}.1-osx64.tar.xz",
```
Comment (samford) [inline Casks/wine-staging.rb:8]:
```suggestion
  url "https://github.com/Gcenx/macOS_Wine_builds/releases/download/#{version.major_minor}/wine-staging-#{version}-osx64.tar.xz",
```

**Review by samford — COMMENTED** [review]
This isn't the first cask I've seen where a release asset contains a more specific version than the tag. [I can't remember which cask it was but I've seen a project add new release assets for a one-off patch release instead of creating a new release.]

This is probably just a fluke for `wine-staging` but we could work around this by using the full version (`version "8.17.1"`) and `version.major_minor` for the tag part in the `url`. If we use the full version from the file name, I'll push an updated `livecheck` block to match the version from the release asset instead (`GithubLatest` checks the release tag by default).

This setup would also work for typical versions where the tag is just the major/minor and the file name version matches (e.g., `8.10`, `wine-staging-8.10-osx86.tar.xz`). However, we would need to modify the `url` to use the full version for the tag again if/when they create a release with a tag like `8.0.1`. [That has only happened a few times in the past, so `version.major_minor` will be fine until then.]

**Comment by miccal** [comment]
Thank you @samford.

**Review by miccal — APPROVED** [review]

**Pull request merged**